staging/src/k8s.io/apiserver/pkg/storage/cacher/time_budget.go

```diff
@@ -48,48 +54,38 @@ type timeBudgetImpl struct {
 	sync.Mutex
 	budget time.Duration
 
-	refresh   time.Duration
 	maxBudget time.Duration
+	clock     clock.Clock
+	limiter   *rate.Limiter
 }
 
-func newTimeBudget(stopCh <-chan struct{}) timeBudget {
+func newTimeBudget() timeBudget {
+	clock := clock.RealClock{}
+	limiter := rate.NewLimiter(rate.Limit(1), burstSize)
 	result := &timeBudgetImpl{
+		clock:     clock,
 		budget:    time.Duration(0),
-		refresh:   refreshPerSecond,
 		maxBudget: maxBudget,
+		limiter:   limiter,
 	}
-	go result.periodicallyRefresh(stopCh)
+	clock.Sleep(limiter.ReserveN(clock.Now(), burstSize).DelayFrom(clock.Now()))
```

Comment:
not sure tokens should be reserved which makes `takeAvailable()` return 0.

Reply:
I don't think that really matters.

IIRC, limiter starts at 0, which would be consistent with behavior so far, so let's delete it maybe?

Reply:
> IIRC, limiter starts at 0, which would be consistent with behavior so far, so let's delete it maybe?

line 37 in time_budget_test.go could prove limiter begin with a full bucket.

Reply:
I'm fine with starting with full buffer too actually - just document it.

Reply:
Yeah, removed.